app/controllers/sign_in/client_configs_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module SignIn
+  class ClientConfigsController < SignIn::ApplicationController
+    rescue_from ActiveRecord::RecordNotFound, with: :not_found
+    skip_before_action :authenticate
+    before_action :authenticate_service_account
+    before_action :set_client_config, only: %i[show update destroy]
+
+    def index
+      client_configs = if params[:client_ids].presence
+                         SignIn::ClientConfig.where(client_id: params[:client_ids])
+                       else
+                         SignIn::ClientConfig.all
+                       end
+
+      render json: client_configs, status: :ok
+    end
+
+    def show
+      render json: @client_config, status: :ok
+    end
+
+    def create
+      client_config = SignIn::ClientConfig.new(client_config_params)
+
+      if client_config.save
+        render json: client_config, status: :created
+      else
+        render json: client_config.errors, status: :unprocessable_entity
+      end
+    end
+
+    def update
+      if @client_config.update(client_config_params)
+        render json: @client_config, status: :ok
+      else
+        render json: @client_config.errors, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      @client_config.destroy
```

Comment:
do we need a status response here? or is :ok inferred?

Reply:
This will render a 204 No Content but we could also do a 200 here. I think either are acceptable here

Reply:
I updated it to be more explicit